src/shared/infra/http/controllers/bank/bank.controller.spec.ts

```diff
@@ -1,64 +1,93 @@
 import { HttpStatus, INestApplication } from '@nestjs/common';
-import { Test } from '@nestjs/testing';
+import { Test, TestingModule } from '@nestjs/testing';
 import * as request from 'supertest';
 
 import { AppModule } from '@/app.module';
 import { IBankRepository } from '@/bank/interfaces';
-import { FindBankByIdUseCase } from '@/bank/use-cases';
+import { FindAllBanksUseCase, FindBankByIdUseCase } from '@/bank/use-cases';
 import { Bank } from '@/bank/infra/entities';
 
 describe('Bank controller E2E', () => {
   let app: INestApplication;
   let bankId: string;
   let nonExistentBankId: string;
   let findBankByIdUseCase: FindBankByIdUseCase;
+  let findAllUseCase: FindAllBanksUseCase;
+  let testModule: TestingModule;
+
+  const mockBank: Bank = {
+    id: 'bank-01',
+    name: 'anyBank',
+    logo_url: 'anyURL',
+  } as Bank;
 
   beforeAll(async () => {
-    const module = await Test.createTestingModule({
+    testModule = await Test.createTestingModule({
       imports: [AppModule],
       providers: [
         {
           provide: IBankRepository,
           useValue: {
             findById: jest.fn(),
+            findAll: jest.fn(),
           },
         },
       ],
     }).compile();
 
-    findBankByIdUseCase = module.get<FindBankByIdUseCase>(FindBankByIdUseCase);
+    findBankByIdUseCase =
+      testModule.get<FindBankByIdUseCase>(FindBankByIdUseCase);
+    findAllUseCase = testModule.get<FindAllBanksUseCase>(FindAllBanksUseCase);
 
     bankId = '87b2a64b-2651-422a-8659-c85fedafdc78';
     nonExistentBankId = 'f632a171-e958-4006-98cc-052cfedb82b5';
 
-    app = module.createNestApplication();
+    app = testModule.createNestApplication();
     await app.init();
   });
 
   afterAll(async () => {
     await app.close();
   });
 
-  describe('/bank/:id (GET)', () => {
-    it('should be able to return an existing bank', async () => {
-      const bank = {
-        bank: {
-          id: bankId,
-          name: 'anyBank',
-          logo_url: 'anyURL',
-        } as Bank,
-      };
+  describe('/bank/all (GET)', () => {
+    it('should be able return all banks', async () => {
+      const banks = [mockBank, mockBank];
+      jest.spyOn(findAllUseCase, 'execute').mockResolvedValue({ banks });
+
+      const response = await request(app.getHttpServer())
+        .get('/bank/all')
+        .expect(HttpStatus.OK);
+
+      expect(response.body).toEqual(banks);
+    });
+
+    it('should return an empty list for a nonexistent bank', async () => {
```

Comment:
Não tem sentido essa descrição.

"Deve retornar uma lista vazia para um banco inexistente"

Acho que seria melhor: "should be able to return an empty list"

Reply:
Aqui eu falaria a mesma coisa que coloquei no comentário acima, a descrição deve explicar o que o teste está validando